canonicalwebteam/search/views.py

```diff
@@ -71,13 +71,28 @@ def search_view():
                 num=num,
             )
 
-        return flask.render_template(
-            template_path,
-            query=query,
-            start=start,
-            num=num,
-            results=results,
-            siteSearch=site_search,
-        )
+            response = flask.make_response(
+                flask.render_template(
+                    template_path,
+                    query=query,
+                    start=start,
+                    num=num,
+                    results=results,
+                    siteSearch=site_search,
+                )
+            )
+            # nofollow, noindex
+            response.headers.set("X-Robots-Tag", "none")
+
+            return response
```

Comment:
FYI you could also write this as:

```suggestion
            return flask.render_template(
                template_path,
                query=query,
                start=start,
                num=num,
                results=results,
                siteSearch=site_search,
            ), {"X-Robots-Tag": "noindex"}
```

See https://flask.palletsprojects.com/en/2.0.x/quickstart/#about-responses:

> If a tuple is returned the items in the tuple can provide extra information. Such tuples have to be in the form (response, status), (response, headers), or (response, status, headers). The status value will override the status code and headers can be a list or dictionary of additional header values.

But it's up to you which you think is nicer.